### PR TITLE
Add callbacks

### PR DESCRIPTION
**Pull request by toddkao**
https://github.com/user-attachments/assets/e8ce9c4b-097b-4fbc-ac78-92df456bab57

### CODE REVIEW

File: packages/widget/src/store/callbacks.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  onTransactionBroadcasted?: (props: OnTransactionBroadcasted) => void;
+  onTransactionComplete?: (props: OnTransactionComplete) => void;
+  onTransactionFailed?: (props: OnTransactionFailed) => void;
+};
+
+export const defaultValues: WalletCallbackStore = {};
+
+export const useCallbackStore = create<WalletCallbackStore>()(
+  () => defaultValues
+);
```
Comment (toddkao) [inline packages/widget/src/store/callbacks.ts:41]: callbacks can go anywhere in the codebase and I don't want to prop drill them everywhere, I think it makes sense to use a store for this

Comment (codingki) [reply packages/widget/src/store/callbacks.ts:41]: yeah looks good

**Review by codingki — APPROVED** [review]
lgtm

**Comment by codingki** [comment]
Please also update the docs

**Pull request merged**